homeassistant/components/watson_tts/tts.py

```diff
@@ -0,0 +1,136 @@
+"""Support for IBM Watson TTS integration."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.tts import PLATFORM_SCHEMA, Provider
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_URL = 'watson_url'
+CONF_APIKEY = 'watson_apikey'
+ATTR_CREDENTIALS = 'credentials'
+
+DEFAULT_URL = 'https://stream.watsonplatform.net/text-to-speech/api'
+
+CONF_VOICE = 'voice'
+CONF_OUTPUT_FORMAT = 'output_format'
+CONF_TEXT_TYPE = 'text'
+
+SUPPORTED_VOICES = [
+    "de-DE_BirgitVoice",
+    "de-DE_BirgitV2Voice",
+    "de-DE_DieterVoice",
+    "de-DE_DieterV2Voice"
+    "en-GB_KateVoice",
+    "en-US_AllisonVoice",
+    "en-US_AllisonV2Voice",
+    "en-US_LisaVoice",
+    "en-US_LisaV2Voice",
+    "en-US_MichaelVoice",
+    "en-US_MichaelV2Voice",
+    "es-ES_EnriqueVoice",
+    "es-ES_LauraVoice",
+    "es-LA_SofiaVoice",
+    "es-US_SofiaVoice",
+    "fr-FR_ReneeVoice",
+    "it-IT_FrancescaVoice",
+    "it-IT_FrancescaV2Voice",
+    "ja-JP_EmiVoice",
+    "pt-BR_IsabelaVoice"
+]
+
+SUPPORTED_OUTPUT_FORMATS = [
+    'audio/flac',
+    'audio/mp3',
+    'audio/mpeg',
+    'audio/ogg',
+    'audio/ogg;codecs=opus',
+    'audio/ogg;codecs=vorbis',
+    'audio/wav'
+]
+
+CONTENT_TYPE_EXTENSIONS = {
+    'audio/flac': 'flac',
+    'audio/mp3': 'mp3',
+    'audio/mpeg': 'mp3',
+    'audio/ogg': 'ogg',
+    'audio/ogg;codecs=opus': 'ogg',
+    'audio/ogg;codecs=vorbis': 'ogg',
+    'audio/wav': 'wav',
+}
+
+DEFAULT_VOICE = 'en-US_AllisonVoice'
+DEFAULT_OUTPUT_FORMAT = 'audio/mp3'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_URL, default=DEFAULT_URL): cv.string,
+    vol.Inclusive(CONF_APIKEY, ATTR_CREDENTIALS): cv.string,
+    vol.Optional(CONF_VOICE, default=DEFAULT_VOICE): vol.In(SUPPORTED_VOICES),
+    vol.Optional(CONF_OUTPUT_FORMAT, default=DEFAULT_OUTPUT_FORMAT):
+        vol.In(SUPPORTED_OUTPUT_FORMATS),
+})
+
+
+def get_engine(hass, config):
+    """Set up IBM Watson TTS component."""
+    from ibm_watson import TextToSpeechV1
+
+    service = TextToSpeechV1(
+        url=config.get(CONF_URL),
```

Comment:
Since we have default value
```suggestion
        url=config[CONF_URL],
```